src/MOI/callbacks/MOI_heuristic.jl

```diff
@@ -0,0 +1,88 @@
+function moi_heuristic_xprs_optnode_wrapper(func::Function, model::Optimizer, callback_data::CD) where {CD<:CallbackData}
+    info = model.callback_table.moi_heuristic::Union{Tuple{CallbackInfo{CD}}, Nothing}
+
+    if !isnothing(info)
+        push_callback_state!(model, CS_MOI_HEURISTIC)
+
+        get_callback_solution!(model, callback_data.node_model)
+
+        # Allow at most one heuristic solution per LP optimal node
+        if Xpress.getintattrib(callback_data.node_model, Xpress.Lib.XPRS_CALLBACKCOUNT_OPTNODE) <= 1
+            func(callback_data)
+        end
+
+        pop_callback_state!(model)
+    end
+
+    return nothing
+end
+
+function MOI.set(model::Optimizer, ::MOI.HeuristicCallback, ::Nothing)
+    info = model.callback_table.moi_heuristic
+
+    if !isnothing(info)
+        xprs_optnode_info, = info
```

Comment:
why?

Reply:
Exaplained below

Reply:
To unregister callbacks

Reply:
I mean, why do you need this line? why no using `model.callback_table.moi_heuristic` directly?

Reply:
OK, fixed